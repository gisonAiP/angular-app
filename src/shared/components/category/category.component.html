<div class="text-center category">
  <img data-test="category-image" [src]="imageURL" alt="search" [height]="height"/>
  <p data-test="category-title" class="small py-2">{{title}}</p>
</div>
