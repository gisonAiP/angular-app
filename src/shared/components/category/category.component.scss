.category {
  padding: 0 9px;
}
