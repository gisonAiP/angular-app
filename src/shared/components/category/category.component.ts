import {Component, Input } from '@angular/core';

@Component({
  selector: 'app-category',
  templateUrl: './category.component.html',
  styleUrls: ['./category.component.scss']
})
export class CategoryComponent {
  @Input() title: string = 'Default';
  @Input() imageURL: string | null = null
  @Input() height = 64
  constructor() { }
}
