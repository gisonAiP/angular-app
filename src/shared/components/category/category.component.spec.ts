import { createRoutingFactory } from '@ngneat/spectator';
import { Spectator } from '@ngneat/spectator/jest';
import { CategoryComponent } from './category.component';


describe('CategoryComponent', () => {
  let spectator: Spectator<CategoryComponent>;
  const createComponent = createRoutingFactory({
    component: CategoryComponent,
    shallow: true,
  });

  beforeEach(() => {
    spectator = createComponent({
      props: {
        title: 'Test title',
        imageURL: 'dummy-url',
        height: 100
      }
    });
  });

  it('should validate snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should display category title', () => {
    expect(spectator.query('[data-test="category-title"]').textContent).toEqual('Test title');
  });

  it('should display image url', () => {
    expect(spectator.query<HTMLImageElement>('[data-test="category-image"]').src).toEqual('http://localhost/dummy-url');
  });

  it('should display image height', () => {
    expect(spectator.query<HTMLImageElement>('[data-test="category-image"]').height).toEqual(100);
  });
});

