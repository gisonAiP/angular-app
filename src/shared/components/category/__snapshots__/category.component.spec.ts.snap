// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryComponent should validate snapshot 1`] = `
<app-category
  height={[Function Number]}
  imageURL={[Function String]}
  title={[Function String]}
>
  <div
    class="text-center category"
  >
    <img
      alt="search"
      data-test="category-image"
      height="100"
      src="dummy-url"
    />
    <p
      class="small py-2"
      data-test="category-title"
    >
      Test title
    </p>
  </div>
</app-category>
`;
